### DISCUSSION TITLE
Does ms build script debugging still works?

### DISCUSSION THREAD

**Discussion by roozbehid** [dotnet/msbuild#10405] [General]
There were some blog post about how to enable `/debug` on ms build and how attaching debugger and stepping into code would result in debugging csproj file.
Also I found more recent posts about setting environment variable that launches then attaches to ms build. I tried all of the and still cannot step into csproj or vcxproj.
Are these solutions still work

**Comment by rainersigwald** [comment]
No, that feature was never made official and has been removed. Logging, and especially [binlogs](https://aka.ms/msbuild/binlog) combined with the [Structured Log Viewer](https://msbuildlog.com) are the preferred debugging approach.

**Reply by roozbehid** [reply]
Sure but not good enough.
I am working on https://github.com/roozbehid/dotnet-vcxproj which enables reusing of vcxproj types for Linux and other compiler, so I do many msbuildish hacks. Problem is there are many properties that somehow popup and or item lists used that I don't know who sets them and bi logs also don't show who sets those properties or items.
Is there any other way to track such things?

**Reply by rainersigwald** [reply]
Yes, that information should be extractable in the logs (we're planning to make it easier with https://github.com/dotnet/msbuild/issues/2711, but it should be doable today). If you can share a concrete example of a problem you're facing I might be able to help?

**Reply by roozbehid** [reply]
Sure. Here I created almost empty `.vcxproj library` and then using msbuild to compile it.
This is the information I am seeing from MSBuild Log Viewer

![image](https://github.com/user-attachments/assets/eb495100-ade1-4175-9ee5-7583e43fdc68)

What I want to know is, what set properties for each `ClCompile`. For example, for `DynamicLib.cpp`, I have `ProgramDataBaseFileName = x64\Debug\vc143.pdb`, I want to know what line of code set this?
